### PR TITLE
Dev varRBE robopt world coordinate system

### PR DESCRIPTION
**Pull request by amitantony**
This PR should replace the old world coordinate system PR #704

### CODE REVIEW

**Comment by amitantony** [comment]
comments for reviewers
1. Ive checked the examples run through , please let me know if there are new parts of code not covered in the examples that ive missed
2. would appreciate Reviewers with more MC experience to test out those parts of the code, results for Example 12 were quite close to that of the clean dev varRBErobOpt but Im unsure of how close, close should be 
cheers 🍻

**Comment by wahln** [comment]
Can you quickly write a summary about
- the different coordinates system now used and where there (default) origin is? I see the world system, which probably is given by ct.x/y/z and is defaulted like it was before.  And I see the cube system which uses the i,j,k and translates directly to coordinates.
- Which entities are living in which coordinate system? Isocenter always in world? doseGrid.x/y/z also in world? When is the cube system used (only internal functions)?
- Maybe it would make sense to already consider the orientation of the system kn the functions somehow. Such that in the future, we could also transform easily between different medical systems / anatomical orientations.

File: matRad/plotting/matRad_plotAxisLabels.m [hunk 1]
```diff
@@ -37,47 +37,55 @@ function  matRad_plotAxisLabels(axesHandle,ct,plane,slice,defaultFontSize,tickdi
     defaultFontSize = 12;
 end
 
+if ~isfield(ct,'x') || isempty(ct.x)
+    ct = matRad_getWorldAxes(ct);
+end
+
```
Comment (wahln) [inline matRad/plotting/matRad_plotAxisLabels.m:43]: See other comments

File: matRad/tools/matRad_generateSingleBixelStf.m [hunk 2]
```diff
@@ -95,18 +95,17 @@
 
 %Get Isocenter voxel as target
 if ~any(isfield(ct,{'x','y','z'}))
-    ct.x = ct.resolution.x*[1:ct.cubeDim(2)]-ct.resolution.x/2;
-    ct.y = ct.resolution.y*[1:ct.cubeDim(1)]-ct.resolution.y/2;
-    ct.z = ct.resolution.z*[1:ct.cubeDim(3)]-ct.resolution.z/2;
+   ct = matRad_getWorldAxes(ct);
+
 end
```
Comment (wahln) [inline matRad/tools/matRad_generateSingleBixelStf.m:100]: See other comments

File: matRad/tools/matRad_generateSingleBixelStf.m [hunk 3]
```diff
@@ -95,18 +95,17 @@
...
 %xVox = ct.x + ct.resolution.x/2;
 %yVox = ct.y + ct.resolution.y/2;
 %zVox = ct.z + ct.resolution.z/2;
 
-xVox = ct.resolution.x*[1:ct.cubeDim(2)]-ct.resolution.x/2;
-yVox = ct.resolution.y*[1:ct.cubeDim(1)]-ct.resolution.y/2;
-zVox = ct.resolution.z*[1:ct.cubeDim(3)]-ct.resolution.z/2;
+% xVox = ct.resolution.x*[1:ct.cubeDim(2)]-ct.resolution.x/2;
+% yVox = ct.resolution.y*[1:ct.cubeDim(1)]-ct.resolution.y/2;
+% zVox = ct.resolution.z*[1:ct.cubeDim(3)]-ct.resolution.z/2;
```
Comment (wahln) [inline matRad/tools/matRad_generateSingleBixelStf.m:108]: If this is no longer needed, let's delete it

File: matRad/tools/matRad_generateSingleBixelStf.m [hunk 4]
```diff
@@ -159,7 +158,10 @@
     stf(i).ray.rayPos_bev = [0 0 0];
     stf(i).ray.targetPoint_bev = [0 SAD 0];
     
-    stf(i).ray.rayPos = stf.isoCenter;
+    
+    shiftedIsoCenter = vertcat(stf(:).isoCenter) - [ct.x(1) ct.y(1) ct.z(1)] + [ct.resolution.x ct.resolution.y ct.resolution.z] ;
```
Comment (wahln) [inline matRad/tools/matRad_generateSingleBixelStf.m:162]: Again this annoying transformation

**Review by wahln — CHANGES_REQUESTED** [review]
Good work, but I think some stuff could be a bit cleaner. Also, there seems to be some regular transformations which could deserve their own functions and are not directly clear why they are done this way (maybe add comments).

File: matRad/geometry/matRad_world2cubeCoords.m [hunk 5]
```diff
@@ -0,0 +1,38 @@
+function coord = matRad_world2cubeCoords(wCoord, ct)
```
[the remark this answers is not in the document]
Comment (amitantony) [reply matRad/geometry/matRad_world2cubeCoords.m:1]: Ive set up a way for us to either pass the ct or dij.doseGrid or the dij.ctGrid. does this make sense ?

[the remark this answers is not in the document]
Comment (wahln) [reply matRad/geometry/matRad_world2cubeCoords.m:1]: Yes. I think that the ctGrid and doseGrid in the dij look quite similar. They also have a .resolution.x/y/z and potentially an .x/y/z with the plan coordinates. The difference to the ct is that ct has "cubeDim" while the grids have "dimensions".

File: matRad/geometry/matRad_projectOnComponents.m [hunk 6]
```diff
@@ -31,7 +31,7 @@
 %
 % Copyright 2017 the matRad development team.
 %
-% This file is not part of the offical matRad release
+% This file is not part of the offical matRad release ?
```
[the remark this answers is not in the document]
Comment (amitantony) [reply matRad/geometry/matRad_projectOnComponents.m:34]: was a personal note, my bad. 
why is this not part of the official release ?

**Comment by amitantony** [comment]
Summary of the different coordinate systems :
1. voxel coordinates ( i, j, k) in cube
used for all handling within the cube and GUI 

2. world coordinates (x, y, z) [mm]
where the isocenter is defined, therefore consistent with all dicom patient positioning 

3. raytracing coordinates, same as world but shifted so that isocenter is at origin (x, y, z)
this is used by raytracing and all the dosecalc Functions

File: matRad/matRad_generateStf.m [hunk 7]
```diff
@@ -218,9 +223,11 @@
...
         % Correct for iso center position. Whit this correction Isocenter is
         % (0,0,0) [mm]
-        coordsX = coordsX_vox*ct.resolution.x - pln.propStf.isoCenter(i,1);
-        coordsY = coordsY_vox*ct.resolution.y - pln.propStf.isoCenter(i,2);
-        coordsZ = coordsZ_vox*ct.resolution.z - pln.propStf.isoCenter(i,3);
+        wCoords = matRad_cube2worldCoords([coordsX_vox, coordsY_vox, coordsZ_vox], ct);
+
+        coordsX = wCoords(:,1) - pln.propStf.isoCenter(i,1);
+        coordsY = wCoords(:,2) - pln.propStf.isoCenter(i,2);
+        coordsZ = wCoords(:,3) - pln.propStf.isoCenter(i,3);
```
[the remark this answers is not in the document]
Comment (amitantony) [reply matRad/matRad_generateStf.m:230]: didnt want to change the coordsX /Y/Z notation, if i change that then its one line

File: matRad/geometry/matRad_projectOnComponents.m (hunk 6 above)
[the remark this answers is not in the document]
Comment (wahln) [reply matRad/geometry/matRad_projectOnComponents.m:34]: This line should be removed and the whole section should be replaced with the standard message.

File: matRad/matRad_generateStf.m (hunk 7 above)
[the remark this answers is not in the document]
Comment (wahln) [reply matRad/matRad_generateStf.m:230]: If we don't need the coords(X/Y/Z) separately further down, it is good to make a one-liner out of this.

**Comment by JenHardt** [comment]
> comments for reviewers
> 
> 1. Ive checked the examples run through , please let me know if there are new parts of code not covered in the examples that ive missed
> 2. would appreciate Reviewers with more MC experience to test out those parts of the code, results for Example 12 were quite close to that of the clean dev varRBErobOpt but Im unsure of how close, close should be
>    cheers 🍻

i think from a MC point of view you are good to go

**Comment by wahln** [comment]
Also there's a huge number of tests failing:
https://github.com/e0404/matRad/actions/runs/10060022215
Unfortunately currently printingthe failures do not work on all PRs, but it will as soon as the new pipeline is on the master-branch.

**Review by wahln — APPROVED** [review]
Looks good now!
However, I wonder how you manually tested the alignment of the grids? Just by looking at the dosimetric results (equivalence) or did you debug and check that similar values are generated?

**Review by wahln — APPROVED** [review]

**Comment by wahln** [comment]
Will take a look on why the tests don't run through.

**Comment by wahln** [comment]
> Will take a look on why the tests don't run through.

My bad, it's only the failing coverage report in PRs.

**Pull request merged**